net2/HostManager.js

```diff
@@ -1411,6 +1431,31 @@ module.exports = class HostManager {
                   }
                 }
               }
+
+              const updatedPolicy = this.policy["vpnClient"];
+              if (settings.strictVPN) {
+                if (updatedPolicy && updatedPolicy.waitresume && updatedPolicy.waitresume == 1 && fc.isFeatureOn("vpn_restore")){
+                  const device_cout = await this.getVpnActiveDeviceCount(profileId);
+                  let alarm = new Alarm.VPNRestoreAlarm(
+                    new Date() / 1000,
+                    null,
+                    {
+                      'p.vpn.profileid': profileId,
+                      'p.vpn.subtype': settings && settings.subtype,
+                      'p.vpn.devicecount': device_cout,
+                      'p.vpn.displayname': (settings && (settings.displayName || settings.serverBoxName)) || profileId,
+                      'p.vpn.time': new Date() / 1000,
+                      'p.vpn.strictvpn': settings && settings.strictVPN || false
+                    }
+                  );
+                  await alarmManager2.enqueueAlarm(alarm);
+                }
+              }
+              if (updatedPolicy) {
+                updatedPolicy.waitresume = 0;
+                updatedPolicy.running = true;
+                await this.setPolicyAsync("vpnClient", updatedPolicy);
```

Comment:
do not call setPolicy in the synchronized part of this function. Return the updated part instead.

Reply:
Otherwise, it may trigger concurrent execution of this function